File: java/lab5/app/src/main/java/com/practice/viewpage/ThirdFragment.java
package com.practice.viewpage;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;


public class ThirdFragment extends Fragment {
    
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.e("ThirdFragment", "onCreate");
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        Log.e("ThirdFragment", "onCreateView");
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_third, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        Log.e("ThirdFragment", "onViewCreated");
    }

    @Override
    public void onStart() {
        super.onStart();
        Log.e("ThirdFragment", "Start");
    }

    @Override
    public void onResume() {
        super.onResume();
        Log.e("ThirdFragment", "Resume");
    }

    @Override
    public void onPause() {
        super.onPause();
        Log.e("ThirdFragment", "Pause");
    }

    @Override
    public void onStop() {
        super.onStop();
        Log.e("ThirdFragment", "onStop");
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Log.e("ThirdFragment", "onDestroy");
    }

    @Override
    public void onDetach() {
        super.onDetach();
        Log.e("ThirdFragment", "onDetach");
    }
}

File: kotlin/lab5/app/src/main/java/com/practice/viewpage/ThirdFragment.kt
package com.practice.viewpage

import android.os.Bundle
import android.util.Log
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

class ThirdFragment : Fragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.e("ThirdFragment", "onCreate")
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        Log.e("ThirdFragment", "onCreateView")
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_third, container, false)
    }

    // onActivityCreated is deprecated
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        Log.e("ThirdFragment", "onViewCreated")
    }
    override fun onStart() {
        super.onStart()
        Log.e("ThirdFragment", "Start")
    }

    override fun onResume() {
        super.onResume()
        Log.e("ThirdFragment", "Resume")
    }

    override fun onPause() {
        super.onPause()
        Log.e("ThirdFragment", "Pause")
    }

    override fun onStop() {
        super.onStop()
        Log.e("ThirdFragment", "onStop")
    }

    override fun onDestroyView() {
        super.onDestroyView()
        Log.e("ThirdFragment", "onDestroyView")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.e("ThirdFragment", "onDestroy")
    }

    override fun onDetach() {
        super.onDetach()
        Log.e("ThirdFragment", "onDetach")
    }
}